src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs

```diff
@@ -506,14 +944,15 @@ private void ScanFormatSpecifier()
                     }
                     else if (ch == '{')
                     {
-                        TrySetUnrecoverableError(_lexer.MakeError(_lexer.TextWindow.Position, 1, ErrorCode.ERR_UnexpectedCharacter, ch));
+                        TrySetError(_lexer.MakeError(
+                            _lexer.TextWindow.Position, 1, ErrorCode.ERR_UnexpectedCharacter, ch));
                         _lexer.TextWindow.AdvanceChar();
                     }
                     else if (ch == '}')
                     {
                         return; // end of interpolation
                     }
-                    else if (IsAtEnd())
+                    else if (IsAtEnd(allowNewline: true))
```

Comment:
Do we have tests for newline in format specifier?

Reply:
Yup.  TrailingSpaceInFormatSpecifier02 tests that.